File: lang/go/test/fun/vac/data_structures/LinkedQueue/LinkedQueueTest.go
package main

import (
    "fun/vac/data_structures/LinkedQueue"
    "fun/vac/util/TestRunner"
)

func testLinkedQueue() bool {
    size := 8192

    queue := LinkedQueue.New()

    for i := 1; i <= size; i++ {
        queue.Enqueue(i)
    }

    if queue.Size() != size {
        return false
    }

    for i := 1; i <= size; i++ {
        if queue.Peek() != i {
            return false
        }
        queue.Dequeue()
    }

    return queue.IsEmpty()
}

func main() {
    TestRunner.ParseTest(testLinkedQueue())
}

File: lang/php/test/vac.fun/data_structures/LinkedQueueTest.php
<?php
    require_once "vac.fun/data_structures/LinkedQueue.php";
    require_once "vac.fun/util/TestRunner.php";

    use vac\fun\data_structures\LinkedQueue\LinkedQueue;
    use vac\fun\util\TestRunner;

    function testLinkedQueue(): bool {
        $size = 8192;

        $queue = new LinkedQueue();

        for ($i = 1; $i <= $size; $i++) {
            $queue->enqueue($i);
        }

        if ($queue->size() != $size) {
            return false;
        }

        for ($i = 1; $i <= $size; $i++) {
            if ($queue->peek() != $i) {
                return false;
            }
            $queue->dequeue();
        }

        return $queue->isEmpty();
    }

    if (count(debug_backtrace()) == 0) {
        TestRunner\parseTest(testLinkedQueue());
    }
?>
